src/test/functional/pages/PimPage.ts

```diff
@@ -1,61 +1,182 @@
+
+  // public getLocatorByRandomNewEmplyeeName(randomNewEmpoyeeName: string):Locator {
+  //   return this.page.locator(`.oxd-table-cell:has(:text("${randomNewEmpoyeeName}"))`);
+  // } this also works, let's keep it in codebase for poor times
   
-    public async searchEmployeeByName(employeeName:string) {
-      await this.employeeNameInput.type(employeeName)
-      await this.searchEmployeeButton.click()
+  public getCellByRandomNewEmplyeeName(
+    randomNewEmployeeName: string
+  ): Locator {
+    return this.page
+    .getByRole('cell', {name: randomNewEmployeeName})
+  }
+
+  // public getTrashBinByRandomEmployeeName(randomNewEmployeeName: string):Locator {
+  //   return this.page.locator(`.oxd-table-card:has(:text("${randomNewEmployeeName}")) .oxd-icon.bi-trash`);
+  // } 
+  // this also works, let's keep it in codebase for poor times
+
+  public getTrashBinByRandomEmployeeName(
+    randomNewEmployeeName: string
+  ): Locator {
+    return this.page
+      .getByText(randomNewEmployeeName)
+      .locator("xpath=../..")
+      .getByRole("button")
+      .locator(".bi-trash");
+  }
+
+  // public  getEditIconByRandomEmployeeName(randomNewEmpoyeeName: string): Locator {
+  //   return this.page.locator( `.oxd-table-card:has(:text("${randomNewEmpoyeeName}")) .oxd-icon.bi-pencil-fill`);
+  // } 
+  //this also works, let's keep it in codebase for poor times
+
+  public getEditIconByRandomEmployeeName(randomNewEmployeeName: string): Locator {
+    return this.page
+    .getByText(randomNewEmployeeName)
+    .locator('xpath=../..')
+    .getByRole('button')
+    .locator('.bi-pencil-fill')
+  }
+
+  // public  getCheckIconByRandomEmployeeName(randomNewEmpoyeeName: string):Locator {
+  //   return this.page.locator(`.oxd-table-card:has(:text("${randomNewEmpoyeeName}")) .oxd-checkbox-input-icon`);
+  // } 
+  // this also works, let's keep it in codebase for poor times
+
+  public getCheckIconByRandomEmployeeName(
+    randomNewEmployeeName: string
+  ): Locator {
+    return this.page
+      .getByText(randomNewEmployeeName)
+      .locator("xpath=../..")
+      .locator("span")
+  }
+
+  public getInputByLabelText(label: string): Locator {
+    return this.page
+      .getByText(label, { exact: true })
+      .locator("xpath=../..")
+      .locator("input");
+  }
+
+  public async addEmployee(firstRandomName: string, bySubtab?: boolean) {
+    if (bySubtab) {
+      await this.navigateToSubPage(SubPage.ADD_EMPLOYEE);
+    } else {
+      await this.getAddEmployeeButtonn().click();
     }
+    await this.getFirstNameInput().type(firstRandomName);
+    await this.getMiddleNameInput().type(newEmployeeTestData.middleName);
+    await this.getLastNameInput().type(newEmployeeTestData.lastName);
+    await this.getSaveUserButton().click();
+  }
 
-    public async searchEmployeeById(employeeId:string) {
-      await this.employeeIdInput.type(employeeId)
-      await this.searchEmployeeButton.click()
+  public async addEmployeeWithLoginDetails(
```

Comment:
"...LoginCredentails" instead of "LoginDetails"? More accurate, but you choice ;)